### DISCUSSION TITLE
External Driving force to flat grain boundary migration

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#26196] [Q&A Modules: Phase field]
Hello All, I want to know if we can add an external driving force to a flat grain boundary in a 2D or 3D space. I am trying to see the interaction of grain boundary and insoluble particles and want my flat grain boundary to move from one one side of the box to other. Is there any example I can look into? Thank you

**Comment by GiudGiud** [comment]
Hello

So this would be a load applied on a surface inside the mesh?
Would you apply it on the sides of elements or on nodes?

Could you use a volumetric load insteasd?

Guillaume

**Reply by ashishdhole** [reply]
I am sorry, I didn't get your question? Can we use a function inside a polycrystalkernel to be acting as an external driving force?

**Reply by GiudGiud** [reply]
This kernel / action ?
https://mooseframework.inl.gov/source/actions/PolycrystalKernelAction.html

**Reply by ashishdhole** [reply]
Yes

**Reply by GiudGiud** [reply]
I dont see an option.
I think you need to look into using kernels individually instead of the action

**Reply by ashishdhole** [reply]
Is there any example I can look into. two grains with individual kernels in normal grain growth model?

**Reply by GiudGiud** [reply]
@amjokisaari @laagesen

**Comment by laagesen** [comment]
Hi @ashishdhole , it might be possible to force a flat GB to move in a simulation, but it would be unphysical. Classical theory of grain boundary growth states that GB velocity is proportional to curvature, see for example the textbook by Porter and Easterling. If there is no curvature, there will be no GB motion. So, we should not expect a flat GB to move. I would instead embed a circular grain inside another grain, then you will get GB motion at a known rate. Have a look here for an example:

https://mooseframework.inl.gov/modules/phase_field/Grain_Growth_Model.html

In the Model Verification section.

**Reply by ashishdhole** [reply]
Thank you @laagesen I understand that the flat grain movement is unphysical, But I have seen papers performing the flat grain movement to see the interaction of grain boundary with particles. (e.g. Tonks, Michael R., Yongfeng Zhang, Aaron Butterfield, and Xian-Ming Bai. "Development of a grain boundary pinning model that considers particle size distribution using the phase field method." Modelling and Simulation in Materials Science and Engineering 23, no. 4 (2015): 045009.) Even if it is unphysical is there a way to do it in moose? 
![2](https://github.com/idaholab/moose/assets/53962301/fddeb241-366d-4718-897f-5384359b961c)
![1](https://github.com/idaholab/moose/assets/53962301/564cc11f-6eb6-47aa-8905-37afc9f22a08)

**Comment by laagesen** [comment]
It looks like this capability was implemented in Marmot, as is stated in the paper, and it looks like it is still in the Marmot repo. However it has not been transferred to MOOSE. Are you affiliated with an institution that has a Marmot license, or could request one?

**Reply by ashishdhole** [reply]
No, My institute do not have Marmot license. we are exploring moose for the first time under the impression that moose can handle large scale simulations easily and less time and computational resources. Flat grain movement and its interactions are the initial tests we want to perform to move further into 3D spaces. Is there any alternate way you can think of? Say using 'BicrystalCircleGrainICAction' and keeping the center of the circular grain out of the mesh such that it still has the curvature but looks flat. "This is just what I can think of right now" Will it work?

**Reply by laagesen** [reply]
I think you could create a circular grain that way and it would move a little bit, but it would soon flatten out and lose its driving force for motion.

**Reply by ashishdhole** [reply]
yes. It is not moving as expected. I think Circular grain in a box is the only thing I can do at this moment. I hope this feature is added to moose in future versions. Thank you all for your prompt replies and help. 
Best Regards
Ashish Dhole
